pypesto/visualize/sampling.py

```diff
@@ -91,7 +117,715 @@ def sampling_fval_trace(
+
+    Parameters
+    ----------
+    levels:
+        The credibility levels.
+    n_variables:
+        The maximum possible number of variables per subplot.
+
+    Returns
+    -------
+    A 2-tuple, with the following indices and values.
+    - `0`: a list of opacities, one per level.
+    - `1`: a list of colors, one per variable.
+    """
+    level_opacities = sorted(
+        # min 30%, max 100%, opacity
+        np.linspace(0.3 * RGBA_MAX, RGBA_MAX, len(levels)),
+        reverse=True,
+    )
+    cmap = plt.cm.viridis
+    cmap_min = RGBA_MIN
+    cmap_max = 0.85*(RGBA_MAX - RGBA_MIN) + RGBA_MIN  # exclude yellows
+
+    # define colormap
+    variable_colors = [
+        list(cmap(v))[:LEN_RGB]
+        for v in np.linspace(cmap_min, cmap_max, n_variables)
+    ]
+
+    return level_opacities, variable_colors
+
+
+def sampling_prediction_trajectories(
+        ensemble_prediction: EnsemblePrediction,
+        levels: Union[float, Sequence[float]],
+        title: str = None,
+        size: Tuple[float, float] = None,
+        axes: matplotlib.axes.Axes = None,
+        labels: Dict[str, str] = None,
+        axis_label_padding: int = 50,
+        groupby: str = CONDITION,
+        condition_gap: float = 0.01,
+) -> matplotlib.axes.Axes:
+    """Plot MCMC-based prediction confidence intervals for the
+    model states or outputs. One or various confidence levels
+    can be depicted. Plots are grouped by condition.
+
+    Parameters
+    ----------
+    result:
+        The pyPESTO result object with filled sample result.
+    levels:
+        Credibility levels, e.g. [95] for a 95% credibility interval. See the
+        :py:func:`_get_level_percentiles` method for a description of how these
+        levels are handled, and current limitations.
+    title:
+        Axes title.
+    size: ndarray
+        Figure size in inches.
+    axes:
+        Axes object to use.
+    labels:
+        Keys should be ensemble output IDs, values should be the desired
+        label for that output. Defaults to output IDs.
+    axis_label_padding:
+        Pixels between axis labels and plots.
+    groupby:
+        Group plots by `pypesto.predict.constants.OUTPUT` or
+        `pypesto.predict.constants.CONDITION`.
+    condition_gap:
+        Gap between conditions when
+        `groupby == pypesto.predict.constants.CONDITION`.
+
+    Returns
+    -------
+    axes:
+        The plot axes.
+    """
+    if labels is None:
+        labels = {}
+    if len(list(levels)) == 1:
+        levels = list(levels)
+    levels = sorted(levels, reverse=True)
+    # Get the percentiles that correspond to the requested credibility levels.
+    percentiles = [
+        percentile
+        for level in levels
+        for percentile in _get_level_percentiles(level)
+    ]
+
+    summary = ensemble_prediction.compute_summary(percentiles_list=percentiles)
+
+    condition_ids, output_ids = _get_condition_and_output_ids(summary)
+
```

Comment:
It would be great to have these id lists also accessible while calling `sampling_prediction_trajectories`.
Something like this:
```python
if condition_ids is None:
   condition_ids, _ = _get_condition_and_output_ids(summary)
if output_ids is None:
    _ , output_ids = _get_condition_and_output_ids(summary)
```
What do you think?